test/kubernetes_unittest.cc

```diff
@@ -1215,4 +1184,705 @@ TEST_F(KubernetesTestFakeServer, MetadataQuery) {
   EXPECT_EQ(pod_metadata->ToString(), m[3].metadata().metadata->ToString());
 }
 
+class KubernetesTestFakeServerOneWatchRetryNodeLevelMetadata
+  : public KubernetesTest {
+ protected:
+  void SetUp() override {
+    server.reset(new testing::FakeServer());
+    KubernetesTest::SetUp();
+  }
+
+  std::unique_ptr<Configuration> CreateConfig() override {
```

Comment:
[Optional] You can add an intermediate class, e.g.:
```c++
class KubernetesTestFakeServerOneWatchRetry : public KubernetesTestFakeServer {
 protected:
  virtual bool ClusterLevel() = 0;
  std::unique_ptr<Configuration> CreateConfig() override {
    return std::unique_ptr<Configuration>(
      new Configuration(std::istringstream(
        "InstanceId: TestID\n"
        "InstanceResourceType: gce_instance\n"
        "InstanceZone: TestZone\n"
        "KubernetesClusterLevelMetadata: " + std::string(ClusterLevel() ? "true" : "false") + "\n"
        "KubernetesClusterLocation: TestClusterLocation\n"
        "KubernetesClusterName: TestClusterName\n"
        "KubernetesEndpointHost: " + server->GetUrl() + "\n"
        "KubernetesNodeName: TestNodeName\n"
        "MetadataIngestionRawContentVersion: TestVersion\n"
        "KubernetesUpdaterWatchConnectionRetries: 1\n"
        "KubernetesUseWatch: true\n"
      )));
  }
};
```
and then your test fixtures become simply:
```c++
class KubernetesTestFakeServerOneWatchRetryNodeLevelMetadata
    : public KubernetesTestFakeServerOneWatchRetry {
 protected:
  bool ClusterLevel() override { return false; }
};

class KubernetesTestFakeServerOneWatchRetryClusterLevelMetadata
    : public KubernetesTestFakeServerOneWatchRetry {
 protected:
  bool ClusterLevel() override { return true; }
};
```

Reply:
Done. (Much better, thanks!)